### ISSUE TITLE
Lb3AppBooterComponent: restRouter mode does not load `loopback.token` middleware

### ISSUE THREAD

**Issue by hacksparrow** [loopbackio/loopback-next#4181]
`Lb3AppBooterComponent` fails to load the `loopback.token` middleware. 

The cause is https://github.com/strongloop/loopback-next/blob/6a972db819aab9f831a26da4f2f81abe3c26efe7/packages/booter-lb3app/src/lb3app.booter.ts#L139

not setting up the `loopback.rest` middleware as expected.

**Comment by bajtos** [comment]
This is working as intended, `restRouter` mode means exactly what it says - we load the LoopBack REST router only, nothing more.

We need to find out how to enable LB3 authentication based on `loopback.token` middleware when using `restRoutes` mode and then update the migration guide for migrating Express middleware.

**Comment by hacksparrow** [comment]
More context.

While working on https://github.com/strongloop/loopback-next/pull/4056, I set the `Lb3AppBooterComponent`'s mode to `restRouter` assuming it would load the `loopback.rest` middleware and the LB3 API's would work as usual.

However, I noticed that  all the LB3 endpoints which uses access tokens stopped working. The middleware defined in `loopback/server/middleware/rest.js` which was called by the LB3 app earlier, was not called anymore, consequently the `loopback.token` middleware is also not loaded, hence none of the tokens are processed.

Compare how the middleware was called in LB3 app:

https://github.com/strongloop/loopback/blob/master/server/middleware/rest.js#L52-L60

and how it is called in `Lb3AppBooterComponent`:

https://github.com/strongloop/loopback-next/blob/master/packages/booter-lb3app/src/lb3app.booter.ts#L139
